### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by munkhuushmgl**
added small formatting and updated public key.

### CODE REVIEW

File: iot/api/CloudIotSample/Program.cs [hunk 1]
```diff
@@ -1376,6 +1491,209 @@ public static object ClearRegistry(string projectId, string cloudRegion, string
         }
         //[END iot_clear_registry]
 
+
+        //[START iot_attach_device]
+        public static object AttachDevice(MqttClient client, string deviceId)
```
Comment (SurferJeffAtGoogle) [inline iot/api/CloudIotSample/Program.cs]: The sample should demonstrate how to create the client.

File: iot/api/CloudIotSample/Program.cs [hunk 2]
```diff
@@ -1376,6 +1491,209 @@ public static object ClearRegistry(string projectId, string cloudRegion, string
...
+            client.Publish(attachTopic, BinaryData, MqttMsgBase.QOS_LEVEL_AT_LEAST_ONCE, true);
+
+            Console.WriteLine("Waiting for device to attach.");
+            System.Threading.Thread.Sleep(2000);
+            return 0;
+        }
+        //[END iot_attach_device]
+
+        //[START iot_detach_device]
+        public static object DetachDevice(MqttClient client, string deviceId)
```
Comment (SurferJeffAtGoogle) [inline iot/api/CloudIotSample/Program.cs]: The sample should demonstrate how to create the client.

File: iot/api/CloudIotSample/Program.cs [hunk 3]
```diff
@@ -1376,6 +1491,209 @@ public static object ClearRegistry(string projectId, string cloudRegion, string
...
+        public static object AttachDevice(MqttClient client, string deviceId)
+        {
+            var attachTopic = $"/devices/{deviceId}/attach";
+            Console.WriteLine("Attaching: {0}", attachTopic);
+            var attachPayLoad = "{}";
+            var BinaryData = Encoding.UTF8.GetBytes(attachPayLoad);
+            client.Publish(attachTopic, BinaryData, MqttMsgBase.QOS_LEVEL_AT_LEAST_ONCE, true);
+
+            Console.WriteLine("Waiting for device to attach.");
+            System.Threading.Thread.Sleep(2000);
```
Comment (SurferJeffAtGoogle) [inline iot/api/CloudIotSample/Program.cs]: Can the code confirm that the device has attached?

File: iot/api/CloudIotTest/CloudIotTest.cs [hunk 4]
```diff
@@ -235,46 +262,118 @@ public void TestListGateways()
...
+                _fixture.RegionId, _fixture.RegistryId,
+            deviceId, gatewayId, _fixture.PrivateKeyPath, "RS256", _fixture.CertPath,
+            "state", "test-message");
+            Assert.Contains("Data sent", sendDataBoundDeviceOut.Stdout);
+            Assert.Contains("On Publish", sendDataBoundDeviceOut.Stdout);
+
+            Assert.DoesNotContain("An error occured", sendDataBoundDeviceOut.Stdout);
+
+            //Clean up
+            Run("unbindDeviceFromGateway", _fixture.ProjectId, _fixture.RegionId,
```
Comment (SurferJeffAtGoogle) [inline iot/api/CloudIotTest/CloudIotTest.cs]: None of this clean up code will run if an assertion fails.  Please add a try {} finally {} block to make sure it always runs.

File: iot/api/CloudIotTest/CloudIotTest.cs [hunk 5]
```diff
@@ -235,46 +262,118 @@ public void TestListGateways()
...
+                _fixture.RegionId, _fixture.RegistryId, gatewayId, deviceId, _fixture.CertPath,
+                _fixture.PrivateKeyPath, "RS256", "--listentime", "10");
+
+            //Assertions
+            Assert.DoesNotContain("error occurred", listenConfigMsgOut.Stdout);
+            Assert.Contains("On Subscribe", listenConfigMsgOut.Stdout);
+            Assert.Contains("On Publish", listenConfigMsgOut.Stdout);
+
+            //Clean up
+            Run("unbindDeviceFromGateway", _fixture.ProjectId, _fixture.RegionId,
```
Comment (SurferJeffAtGoogle) [inline iot/api/CloudIotTest/CloudIotTest.cs]: None of this clean up code will run if an assertion fails.  Please add a try {} finally {} block to make sure it always runs.

I see this pattern repeats.  Consider using a pattern like this:
https://xunit.github.io/docs/shared-context

File: iot/api/CloudIotSample/Program.cs [hunk 6]
```diff
@@ -1376,6 +1491,209 @@ public static object ClearRegistry(string projectId, string cloudRegion, string
...
+                Console.WriteLine("Listening...");
+                var secSinceIssue = SystemClock.Instance.GetCurrentInstant().Minus(jwtIatTime);
+                if (secSinceIssue.TotalSeconds > (60 * jwtExpiresMinutes)
+                    || !mqttClient.IsConnected)
+                {
+                    Console.WriteLine("Refreshing token after {0}s", secSinceIssue);
+                    jwtIatTime = SystemClock.Instance.GetCurrentInstant();
+                    mqttClient = CloudIotMqttExample.GetClient(projectId, cloudRegion,
+                        registryId, deviceId, privateKeyFile, algorithm,
+           caCerts, mqttBridgeHostname, mqttBridgePort);
```
Comment (gguuss) [inline iot/api/CloudIotSample/Program.cs]: This continuation looks a little funny to me, maybe insert some whitespace before `caCerts` to match the previous line

File: iot/api/CloudIotSample/Program.cs [hunk 7]
```diff
@@ -1376,6 +1491,209 @@ public static object ClearRegistry(string projectId, string cloudRegion, string
         }
         //[END iot_clear_registry]
 
+
+        //[START iot_attach_device]
+        public static object AttachDevice(MqttClient client, string deviceId)
+        {
+            var attachTopic = $"/devices/{deviceId}/attach";
+            Console.WriteLine("Attaching: {0}", attachTopic);
+            var attachPayLoad = "{}";
```
Comment (gguuss) [inline iot/api/CloudIotSample/Program.cs]: In the other samples, we added the authorization JWT payload that is [passed to the function by the caller](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/iot/api-client/mqtt_example/cloudiot_mqtt_example.py#L184).

File: iot/api/CloudIotSample/Program.cs (hunk 2 above)
Comment (munkhuushmgl) [reply iot/api/CloudIotSample/Program.cs]: These changes have been made in previous PR which is still under review.

**Pull request merged**